### ISSUE TITLE
MSAL AcquiretokenSilent success case taking 1+ second

### ISSUE THREAD

**Issue by MarkZuber** [AzureAD/microsoft-authentication-library-for-dotnet#1040]
**Which Version of MSAL are you using ?**
MSAL 3.0.1-preview

**Platform**
net45, netcore

The code below is consistently causing 1-1.2 seconds of time for acquire token silent.  It is expected that this should take 5-10ms.

```
for (int i = 0; i < 5000; i++)
{
  Console.Write($"Attempt {i}: ");
  var start = DateTime.Now;
  var result2 = await app.AcquireTokenSilentAsync(scopes, account).ConfigureAwait(false);
  var length = DateTime.Now - start;
  Console.WriteLine($"{length.TotalMilliseconds}ms");
}
```

![msalatsperf](https://user-images.githubusercontent.com/7103178/55520368-d8f71080-5630-11e9-9fdc-e46c5eb41747.png)

**Comment by MarkZuber** [comment]
This is a bug in the test, but we should see if there's a way for us to detect it.  The stress test was sending in https://management.core.windows.net//.default (note the double slash before default) so the cache lookup filtering by scopes was failing.   and then we were doing a full AT token exchange which is where the time came from.  The server ends up taking the // in the scope and normalizing it and when we cache the token, we cache it with /.default as expected (which is where the later lookup mismatch occurs).  So perhaps we should see if we can parse the scopes and if they start with https ensure they're a proper url?

**Comment by jmprieur** [comment]
This will be fixed on the service side (ETA beginning of May)

**Comment by jmprieur** [comment]
@MarkZuber : any objection to closing it?

**Comment by MarkZuber** [comment]
I'd like to keep this active for tracking until we verify the server has fixed this issue.

**Comment by jmprieur** [comment]
then I get I'll move it to 3.0.5+

**Comment by stevehurcombe** [comment]
@MarkZuber do you have any idea when this will get fixed? This is blocking us.

**Comment by jmprieur** [comment]
@MarkZuber : apprently the change should be in prod now.
do you still see the issue?

**Comment by stevehurcombe** [comment]
@jmprieur My path is a bit different. 
I've been lurking here waiting for the right time to upgrade to version 3.x. I saw this issue and I'm following it as I'm seeing the same issue in 2.7.1. A "long" delay accessing the token on every startup.  

@MarkZuber implied that this was a server side issue so I did expect 2.7.1 to benefit as well.

Do we also need to upgrade to version 3.0.8?

Cheers
Steve
